### PR TITLE
flatten cookies for header

### PR DESCRIPTION
**Pull request by tflori**
Fix issue #77

### CODE REVIEW

**Comment by jaapio** [comment]
Could you provide a unittest for this?

**Comment by tflori** [comment]
@jaapio I created a test can you please recheck?

**Comment by jaapio** [comment]
I can. But I won't be able to approve your pr since I'm not a zf
maintainer.

Op do 8 dec. 2016 13:15 schreef tflori <notifications@github.com>:

**Comment by tflori** [comment]
@weierophinney it would be greate if you could review this pr. thanks in advance.

File: src/Header/Cookie.php [hunk 1]
```diff
@@ -97,13 +97,30 @@ public function getFieldValue()
...
-        foreach ($this as $name => $value) {
+        $cookies = [];
+        foreach ($this->flattenCookies($cookies, $this) as $name => $value) {
             $nvPairs[] = $name . '=' . (($this->encodeValue) ? urlencode($value) : $value);
         }
 
         return implode('; ', $nvPairs);
     }
 
+    protected function flattenCookies(&$result, $data, $prefix = null)
```
Comment (ezimuel) [inline src/Header/Cookie.php]: Why we need a recursive function here? Do we need to manage array of array values in cookies? I think we should simplify the implementation here, for instance just iterating on keys and values of $value, if it is an array.

Comment (ezimuel) [inline src/Header/Cookie.php]: Btw, we discourage the usage of passing value by reference in function. A good use case for reference is performance reason, but this is not the case.

Comment (tflori) [inline src/Header/Cookie.php]: For me the correct way seems to be what php is doing. So yes we need a recursive function. But you are right we can write it without passing by reference.

**Comment by tflori** [comment]
@ezimuel I modified the function not to use pass by reference.

I don't know why coveralls shows a decreased coverage status - the function I added is tested. Nothing else changed.

**Comment by ezimuel** [comment]
@tflori thanks for your contribution! I improved the unit test with a bi-dimensional cookie array.

**Pull request merged**